core/backend/src/rpc-impl/IModelReadRpcImpl.ts

```diff
@@ -32,52 +29,46 @@ export class IModelReadRpcImpl extends RpcInterface implements IModelReadRpcInte
   public static register() { RpcManager.registerImpl(IModelReadRpcInterface, IModelReadRpcImpl); }
 
   public async openForRead(tokenProps: IModelRpcOpenProps): Promise<IModelConnectionProps> {
-    const requestContext = ClientRequestContext.current as AuthorizedClientRequestContext;
-    return RpcBriefcaseUtility.openWithTimeout(requestContext, tokenProps, SyncMode.FixedVersion);
+    const user = RpcInvocation.currentRequest as AuthorizedClientRequestContext;
+    return RpcBriefcaseUtility.openWithTimeout(user, tokenProps, SyncMode.FixedVersion);
   }
 
   public async close(_tokenProps: IModelRpcProps): Promise<boolean> {
     return true;
   }
 
   public async queryRows(tokenProps: IModelRpcProps, ecsql: string, bindings?: any[] | object, limit?: QueryLimit, quota?: QueryQuota, priority?: QueryPriority, restartToken?: string, abbreviateBlobs?: boolean): Promise<QueryResponse> {
-    const requestContext = ClientRequestContext.current as AuthorizedClientRequestContext;
-    const iModelDb: IModelDb = await RpcBriefcaseUtility.findOrOpen(requestContext, tokenProps, SyncMode.FixedVersion);
-    requestContext.enter();
-    return iModelDb.queryRows(ecsql, bindings, limit, quota, priority, restartToken, abbreviateBlobs);
+    const user = RpcInvocation.currentRequest as AuthorizedClientRequestContext;
+    const iModelDb = await RpcBriefcaseUtility.findOrOpen(user, tokenProps, SyncMode.FixedVersion);
+    return iModelDb.queryRows(user.sessionId, ecsql, bindings, limit, quota, priority, restartToken, abbreviateBlobs);
```

Comment:
Here's where `queryRows` from an RPC client happens. It gets the sessionId from the "user" so it will be unique across multiple clients, but constant for a given client.